File: client/app/src/main/java/com/qubacy/geoqq/data/mate/request/repository/source/network/model/common/MateRequest.kt
package com.qubacy.geoqq.data.mate.request.repository.source.network.model.common

import com.qubacy.geoqq.data.mate.request.model.DataMateRequest
import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
class MateRequest(
    val id: Long,
    @Json(name = "user-id") val userId: Long
) {

}

fun MateRequest.toDataMateRequest(): DataMateRequest {
    return DataMateRequest(id, userId)
}

File: server/own_services/geoqq_ws/internal/adapters/interfaces/wsApi/internal/dto/serverSide/payload/mateRequest.go
package payload

type MateRequest struct {
	Id     float64 `json:"id"`
	UserId float64 `json:"user-id"`
}

func MakeMateRequest(id, userId float64) MateRequest {
	return MateRequest{
		Id:     id,
		UserId: userId,
	}
}
